ember-core/shared/src/test/scala/org/http4s/ember/core/ParserSuite.scala

```diff
@@ -364,6 +364,39 @@ class ParsingSuite extends Http4sSuite {
     }
   }
 
+  test(
+    "Parser.Response.parser return everything if the body is discarded"
```

Comment:
What happens to the connection in this case?  Discarding an ad hoc number of bytes and then continuing to parse makes me nervous about smuggling attacks.

Reply:
This path is only for response parsing, right?

Reply:
Is the expectation that the connection gets closed, even after a properly framed HEAD response?